From 0665a215d75206f7631a28d5ddca63042bd7b8e7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Wed, 22 May 2024 14:41:30 +0200
Subject: [PATCH 605/954] feat(stm32mp21): add PKA node in STM32MP21 platforms
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch add the node of PKA peripheral that is available for
STM32MP21xF and STM32MP21xC.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 8017e18ef61608ad77ad8f635ca35614970efe84 feat(stm32mp21): add PKA node in STM32MP21 platforms]
Change-Id: Ic30e5dd8686fc849756b43ea390b84741e4a1226
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/381668
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 fdts/stm32mp21xc.dtsi | 8 ++++++++
 fdts/stm32mp21xf.dtsi | 8 ++++++++
 2 files changed, 16 insertions(+)

diff --git a/fdts/stm32mp21xc.dtsi b/fdts/stm32mp21xc.dtsi
index 3aa3e7e3d9..700d399101 100644
--- a/fdts/stm32mp21xc.dtsi
+++ b/fdts/stm32mp21xc.dtsi
@@ -13,5 +13,13 @@
 			resets = <&rcc SAES_R>;
 			status = "disabled";
 		};
+
+		pka: pka@42032000 {
+			compatible = "st,stm32mp13-pka64";
+			reg = <0x42032000 0x2000>;
+			clocks = <&rcc CK_BUS_PKA>;
+			resets = <&rcc PKA_R>;
+			status = "disabled";
+		};
 	};
 };
diff --git a/fdts/stm32mp21xf.dtsi b/fdts/stm32mp21xf.dtsi
index 3aa3e7e3d9..700d399101 100644
--- a/fdts/stm32mp21xf.dtsi
+++ b/fdts/stm32mp21xf.dtsi
@@ -13,5 +13,13 @@
 			resets = <&rcc SAES_R>;
 			status = "disabled";
 		};
+
+		pka: pka@42032000 {
+			compatible = "st,stm32mp13-pka64";
+			reg = <0x42032000 0x2000>;
+			clocks = <&rcc CK_BUS_PKA>;
+			resets = <&rcc PKA_R>;
+			status = "disabled";
+		};
 	};
 };
-- 
2.53.0

